samza-sql-shell/src/main/java/org/apache/samza/sql/client/impl/SamzaExecutor.java

```diff
@@ -0,0 +1,582 @@
+import org.apache.samza.config.JobConfig;
+import org.apache.samza.config.JobCoordinatorConfig;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.config.TaskConfig;
+import org.apache.samza.container.grouper.task.SingleContainerGrouperFactory;
+import org.apache.samza.job.ApplicationStatus;
+import org.apache.samza.serializers.StringSerdeFactory;
+import org.apache.samza.sql.avro.AvroRelSchemaProvider;
+import org.apache.samza.sql.client.interfaces.ExecutionContext;
+import org.apache.samza.sql.client.interfaces.ExecutionStatus;
+import org.apache.samza.sql.client.interfaces.NonQueryResult;
+import org.apache.samza.sql.client.interfaces.QueryResult;
+import org.apache.samza.sql.client.interfaces.SqlExecutor;
+import org.apache.samza.sql.client.interfaces.SqlFunction;
+import org.apache.samza.sql.client.interfaces.SqlSchema;
+import org.apache.samza.sql.client.interfaces.SqlSchemaBuilder;
+import org.apache.samza.sql.client.util.RandomAccessQueue;
+import org.apache.samza.sql.dsl.SamzaSqlDslConverter;
+import org.apache.samza.sql.dsl.SamzaSqlDslConverterFactory;
+import org.apache.samza.sql.fn.FlattenUdf;
+import org.apache.samza.sql.fn.RegexMatchUdf;
+import org.apache.samza.sql.impl.ConfigBasedIOResolverFactory;
+import org.apache.samza.sql.impl.ConfigBasedUdfResolver;
+import org.apache.samza.sql.interfaces.RelSchemaProvider;
+import org.apache.samza.sql.interfaces.RelSchemaProviderFactory;
+import org.apache.samza.sql.interfaces.SqlIOConfig;
+import org.apache.samza.sql.interfaces.SqlIOResolver;
+import org.apache.samza.sql.runner.SamzaSqlApplicationConfig;
+import org.apache.samza.sql.runner.SamzaSqlApplicationRunner;
+import org.apache.samza.sql.testutil.JsonUtil;
+import org.apache.samza.standalone.PassthroughJobCoordinatorFactory;
+import org.apache.samza.system.OutgoingMessageEnvelope;
+import org.apache.samza.system.kafka.KafkaSystemFactory;
+import org.apache.samza.tools.avro.AvroSchemaGenRelConverterFactory;
+import org.apache.samza.tools.avro.AvroSerDeFactory;
+import org.apache.samza.tools.json.JsonRelConverterFactory;
+import org.apache.samza.tools.schemas.ProfileChangeEvent;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.common.base.Joiner;
+import scala.collection.JavaConversions;
+
+import static org.apache.samza.sql.runner.SamzaSqlApplicationConfig.*;
+
+
+public class SamzaExecutor implements SqlExecutor {
+    private static final Logger LOG = LoggerFactory.getLogger(SamzaExecutor.class);
+
+    private static final String SAMZA_SYSTEM_LOG = "log";
+    private static final String SAMZA_SYSTEM_KAFKA = "kafka";
+    private static final String SAMZA_SQL_OUTPUT = "samza.sql.output";
+    private static final String SAMZA_SQL_SYSTEM_KAFKA_ADDRESS = "samza.sql.system.kafka.address";
+    private static final String DEFAULT_SERVER_ADDRESS = "localhost:2181";
+    private static final int RANDOM_ACCESS_QUEUE_CAPACITY = 5000;
```

Comment:
What is this queue capacity? Can you add more details?

Reply:
When user executes a SELECT statement, one of the ways we show data is the logging view, in which new rows of data keep scrolling. However user is able to pause the scrolling view and this capacity is the maximum number of rows we keep background when data accumulates.

Reply:
I added some comment in the code as well.